### PR TITLE
Issue #2 - Further refactor the map related classes into MVVM architecture

### PR DESCRIPTION
**Pull request by KhaledBadran**
#2 Development:

This PR builds upon our attempts to model our application into an MVVM architecture by refactoring the map related classes (and especially the MapFragment). The following points detail the changes that this PR introduces:

**1) Refactor the way we handle buildings images:**

   - The Building model has been updated to include an extra variable that stores the resource id of the building image.
   - Updated map repository to initialize the resource id of the buildings images when initializing the campus buildings.
   - Refactored map fragment by moving the logic to retrieve the resource id of the buildings images to the map viewModel

### 
**2) Refactor the traversal and storing of the campus buildings list:**

   - Both map repository and viewModel have been updated to store a list of all Concordia buildings as a class variable. This is to ensure that the list of buildings is only created once (through campus buildings traversal) and that the logic to create this list of all buildings is the responsibility of map repository only.

### 
**3) Replace hard coded campuses coordinate with external file reading:**

   - Update external buildings.json with coordinates for both campuses
   - Update external buildings.json to be up to date with the JTs branch (highlight remaining buildings) so that future merge conflicts can be avoided.
   - Refactor Map Repository to read campus coordination from the external buildings.json file rather than hard coding those coordinates in the map repository class. 

###
**4) Refactor map drawings of polygon, markers and paths:**

   - The map model now handles the drawing of the buildings polygons and markers on the map 
    and reflects a singleton design pattern so that the map is only initialized once.
   - Improved performance because the map is only initialized once.
   - Refactor the mapping of the building name to the marker resource id by moving it from the map fragment to the map repository.
   - The Building model now stores the resource id for the building marker.
   - Refactor the map fragment to correctly handle the displaying of buildings markers when the app is switched (moved to background).
- Minor update in the Android Manifest xml file for the api key variable name.

### CODE REVIEW

File: app/src/main/java/com/droidhats/campuscompass/repositories/MapRepository.kt [hunk 1]
```diff
@@ -204,12 +204,42 @@ class MapRepository(applicationContext: Context) {
             "Faubourg Building" -> R.drawable.building_fg
             "Visual Arts Building" -> R.drawable.building_va
             "Pavillion J.W. McConnell Building" -> R.drawable.building_webster_library
+            "Grey Nuns Building" -> R.drawable.building_grey_nuns
+            "Samuel Bronfman Building" -> R.drawable.building_sb
+            "GS Building" -> R.drawable.building_gs
+            "CB Building" -> R.drawable.building_cbb
```
Comment (JeremiahTee) [inline app/src/main/java/com/droidhats/campuscompass/repositories/MapRepository.kt]: This should be  R.drawable.building_cb
(remove the last 'b' letter)

**Comment by JeremiahTee** [comment]
I ran the app, tested it manually, ran the tests. I also verified all the commits. Everything checks out. Left minor comments for you to address.

Really great work! Honestly, this refactoring is really inspiring. I'm looking at clean code that I initially didn't know how to properly structure. I am definitely learning a lot from this 🥇

File: app/src/main/java/com/droidhats/campuscompass/repositories/MapRepository.kt (hunk 1 above)
Comment (KhaledBadran) [reply app/src/main/java/com/droidhats/campuscompass/repositories/MapRepository.kt]: Thanks, I've updated the resource name 😁

**Comment by JeremiahTee** [comment]
I have built upon this branch with highlightRemaningBuildings. I fixed some of the errors in this PR (related to buildings), as soon as this PR is merged, I will make a PR for my branch - so that master is 100% stable

Otherwise, amazing work! The singleton is a beast tamed in 039b5bc 😆

**Pull request merged**